main.tf

```diff
@@ -77,8 +76,10 @@ module "app_aks" {
 }
 
 locals {
-  blob_container = local.create_blob_container ? "${module.storage.0.account.name}/${module.storage.0.container.name}" : var.blob_container
-  queue          = var.use_internal_queue ? "internal://" : "az://${module.storage.0.account.name}/${module.storage.0.queue.name}"
+  blob_container  = local.create_blob_container ? module.storage.0.container.name : var.blob_container
+  storage_account = local.create_blob_container ? module.storage.0.account.name : var.storage_account
+  storage_key     = local.create_blob_container ? module.storage.0.account.primary_access_key : var.storage_key
+  queue           = var.use_internal_queue ? "internal://" : "az://${module.storage.0.account.name}/${module.storage.0.queue.name}"
```

Comment:
If the `blob_container` is set then the `create_blob_container` will not be empty which will indicate that we want to use the BYBO, however, if the `use_internal_queue` to `true` along with the `blob_container`, `storage_account` and `storage_key`, the queue creation will fail because it expects that we use the `module.storage` to create the storage account and not bring ours.

Reply:
Here the `use_internal_queue` refers to our queuing solution inside our docker container. So when `use_internal_queue` is set to `true`, then it just uses the `internal://` queue and doesn't need to create one.

Reply:
Actually I meant `false`